### PR TITLE
Convert `commercial` and `commercial.dcr` to Typescript

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Converts the commercial bootstraps to Typescript.

## Does this change need to be reproduced in dotcom-rendering ?

- [X] No -- the bundle will be automatically sent through
- [ ] Yes (please indicate your plans for DCR Implementation)

## What is the value of this and can you measure success?

Adding types helps us identify oddities, remove unused code and be safer.

### Tested

- [X] Locally
- [x] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/bootstraps/commercial.dcr.ts [hunk 1]
```diff
@@ -63,18 +61,28 @@ if (!commercialFeatures.adFree) {
 	);
 }
 
-const loadHostedBundle = () => {
+const loadHostedBundle = (): Promise<void> => {
 	if (config.get('page.isHosted')) {
-        amIUsed('commercial.dcr.js', 'loadHostedBundle', { isHosted: 'true'})
+		amIUsed('commercial.dcr.js', 'loadHostedBundle', { isHosted: 'true' });
```
Comment (chrislomaxjones) [inline static/src/javascripts/bootstraps/commercial.dcr.ts:66]:
```suggestion
		amIUsed('commercial.dcr.ts', 'loadHostedBundle', { isHosted: 'true' });
```

File: static/src/javascripts/bootstraps/commercial.ts [hunk 2]
```diff
@@ -60,17 +60,27 @@ if (!commercialFeatures.adFree) {
...
-					const initHostedVideo = require('commercial/modules/hosted/video');
-					const hostedGallery = require('commercial/modules/hosted/gallery');
-					const initHostedCarousel = require('commercial/modules/hosted/onward-journey-carousel');
-					const loadOnwardComponent = require('commercial/modules/hosted/onward');
+					/* eslint-disable
+					 	@typescript-eslint/no-var-requires,
+						@typescript-eslint/consistent-type-imports,
+						--
+						these are chunked by webpack */
+					const hostedAbout = require('commercial/modules/hosted/about') as typeof import('commercial/modules/hosted/about');
```
Comment (chrislomaxjones) [inline static/src/javascripts/bootstraps/commercial.ts:74]: Not sure about this but is the synchronous equivalent of doing

```suggestion
					const hostedAbout = await import('commercial/modules/hosted/about');
```

**Review by chrislomaxjones — APPROVED** [review]
+1, just two not-so-important comments

File: static/src/javascripts/bootstraps/commercial.ts (hunk 2 above)
Comment (mxdvl) [reply static/src/javascripts/bootstraps/commercial.ts:74]: Yes, I think so. `require.ensure` does not allow for an async function as a second argument, sadly.

Comment (chrislomaxjones) [reply static/src/javascripts/bootstraps/commercial.ts:74]: Gotcha

**Pull request merged**